### PR TITLE
Zoltan:  remove warnings generated by gcc 7.2

### PR DESCRIPTION
**Pull request by kddevin**
@trilinos/zoltan 

## Description

Changes to remove warnings generated when compiling with gcc 7.2.
Most were sprintf to insufficiently allocated strings, or replacing a multiplication with logical-and in if conditions.

## Motivation and Context
SST is moving to gcc 7.3, and reported warnings during their conversion

## How Has This Been Tested?

Tested on skybridge using gcc 7.2 (I couldn't find gcc 7.3 in SEMS).

### CODE REVIEW

File: packages/zoltan/src/driver/dr_hg_io.c [hunk 1]
```diff
@@ -154,7 +154,7 @@ int read_hypergraph_file(
 {
   /* Local declarations. */
   const char  *yo = "read_hypergraph_file";
-  char   cmesg[256];
+  char   cmesg[FILENAME_MAX+256];
```
Comment (mhoemmen) [inline packages/zoltan/src/driver/dr_hg_io.c]: https://www.gnu.org/software/libc/manual/html_node/Limits-for-Files.html

"Usage Note: Don’t use `FILENAME_MAX` as the size of an array in which to store a file name! You can’t possibly make an array that big! Use dynamic allocation (see Memory Allocation) instead."

File: packages/zoltan/src/driver/dr_hg_io.c [hunk 2]
```diff
@@ -166,7 +166,7 @@ int read_hypergraph_file(
   float *hewgts = NULL, *vwgts = NULL;
   ZOLTAN_FILE* fp = NULL;
   int base = 0;   /* Smallest vertex number; usually zero or one. */
-  char filename[256];
+  char filename[FILENAME_MAX+9];
```
Comment (mhoemmen) [inline packages/zoltan/src/driver/dr_hg_io.c]: See above.

File: packages/zoltan/src/driver/dr_hg_io.c [hunk 3]
```diff
@@ -550,7 +550,7 @@ int read_mtxplus_file(
 {
   /* Local declarations. */
   const char  *yo = "read_mtxplus_file";
-  char filename[256], cmesg[256];
+  char filename[FILENAME_MAX+9], cmesg[FILENAME_MAX+256];
```
Comment (mhoemmen) [inline packages/zoltan/src/driver/dr_hg_io.c]: See above.

File: packages/zoltan/src/driver/dr_random_io.c [hunk 4]
```diff
@@ -117,7 +117,7 @@ int create_random_triangles(
 
   short *assignments = NULL;
 
-  char filename[256];
+  char filename[FILENAME_MAX+9];
```
Comment (mhoemmen) [inline packages/zoltan/src/driver/dr_random_io.c]: See above.

File: packages/zoltan/src/driver/dr_random_io.c [hunk 5]
```diff
@@ -314,7 +314,7 @@ int create_random_input(
...
```
Comment (mhoemmen) [inline packages/zoltan/src/driver/dr_random_io.c]: See above.

**Comment by kddevin** [comment]
Since FILENAME_MAX is used only in Zoltan’s test code, not in library code that applications would access, I am not very concerned at this time about its usage.

**Comment by mhoemmen** [comment]
@kddevin wrote:
> Since FILENAME_MAX is used only in Zoltan’s test code, not in library code that applications would access, I am not very concerned at this time about its usage.

I was just thinking that the test might crash, if `FILENAME_MAX` is a large number and there's not enough room on the stack for it.  Stack allocations may be more limited than heap allocations.  If you're OK with it, though, I'll be happy to change my review.

**Comment by kddevin** [comment]
We have been using FILENAME_MAX for years to accommodate nemesis filenames; these changes were just to make sure we didn't copy them into too-small buffers.  If problems arise later, we can change all instances.  Thanks.

**Comment by kddevin** [comment]
@egboman, there no need to review this PR; @mhoemmen chose to do it.  Thanks!  Merging now.

**Pull request merged**